<a href="https://colab.research.google.com/github/ProgramCrafter/colab-foxford-bigdata/blob/master/%D0%9A%D0%BE%D1%80%D0%BE%D1%82%D0%BA%D0%BE%D0%B2%D0%B0_%22%D0%97%D0%B0%D0%BD%D1%8F%D1%82%D0%B8%D0%B5_%E2%84%963_ipynb%22.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/95804/pre3.jpg"></p>

___

<font color="#000000">Основы Python и глубокого обучения</font>
<font color="#AEBAC5"> → </font>
<font color="#AEBAC5">Компьютерное зрение (CV)</font>
<font color="#AEBAC5"> → </font>
<font color="#AEBAC5">Обработка естественного языка (NLP)</font>

___

<h1><strong><font color="#FF732B">Набор данных</font></strong></h1>

| Стоимость, млн $ | Название | Художник | Год продажи | Аукционный дом |
--- | --- | --- | --- | ---
| 478.1 | Спаситель мира | Леонардо да Винчи 🇮🇹 | 2017 | Christie’s |
| 194.5 | Алжирские женщины («Версия O») | Пабло Пикассо 🇪🇸 | 2015 | Christie’s |
| 184.8 | Обнаженная, лежащая на левом боку | Амедео Модильяни 🇮🇹 | 2015 | Christie’s |
| 163.0 | Лежащая обнажённая | Амедео Модильяни 🇮🇹 | 2018 | Sotheby’s |
| 149.6 | Двенадцать ландшафтных сцен | Ци Байши 🇨🇳 | 2017 | Другой |
| 141.9 | Три наброска к портрету Люсьена Фрейда | Фрэнсис Бэкон 🇬🇧 | 2013 | Christie’s |
| 121.3 | Крик, 1895 | Эдвард Мунк 🇳🇴 | 2012 | Sotheby’s |
| 119.2 | Молодая девушка с цветочной корзиной | Пабло Пикассо 🇪🇸 | 2018 | Christie’s |
| 117.3 | Без названия (1982) | Жан-Мишель Баския 🇺🇸 | 2017 | Sotheby’s |
| 112.7 | Стога сена | Клод Моне 🇫🇷 | 2019 | Sotheby’s |
| 109.7 | Дора Маар с кошкой | Пабло Пикассо 🇪🇸 | 2006 | Sotheby’s |
| 107.7 | Обнажённая, зелёные листья и бюст | Пабло Пикассо 🇪🇸 | 2010 | Christie’s |
| 105.0 | Авария серебряной машины (Двойная катастрофа) | Энди Уорхол 🇺🇸 | 2013 | Sotheby’s |
| 103.4 | Медсестра | Рой Лихтенштейн 🇺🇸 | 2015 | Christie’s |
| 101.3 | Портрет Адели Блох-Бауэр II | Густав Климт 🇦🇹 | 2006 | Christie’s |
| 95.2 | Чоп суэй | Эдвард Хоппер 🇺🇸 | 2018 | Christie’s |
| 93.1 | Триптих, 1976 | Фрэнсис Бэкон 🇬🇧 | 2008 | Sotheby’s |
| 90.4 | Буффало 2 | Роберт Раушенберг 🇺🇸 | 2019 | Christie’s |
| 89.9 | Авария зелёной машины | Энди Уорхол 🇺🇸 | 2007 | Christie’s |
| 88.3 | Стог сена | Клод Моне 🇫🇷 | 2016 | Christie’s |
| 87.8 | Оранжевый, красный, жёлтый | Марк Ротко 🇺🇸 | 2012 | Christie’s |
| 81.6 | Белый центр (жёлтое, розовое и лиловое на розовом) | Марк Ротко 🇺🇸 | 2007 | Sotheby’s |
| 81.3 | Черный огонь I | Барнетт Ньюмен 🇺🇸 | 2014 | Christie’s |
| 79.2 | Тройной Элвис | Энди Уорхол 🇺🇸 | 2014 | Christie’s |
| 78.1 | Три наброска к портрету Джона Эдвардса | Фрэнсис Бэкон 🇬🇧 | 2014 | Christie’s |
| 75.9 | № 1 (Королевский красный и голубой) | Марк Ротко 🇺🇸 | 2012 | Sotheby’s |
| 71.9 | Без названия XXV | Виллем де Кунинг 🇺🇸 | 2016 | Christie’s |
| 69.7 | Обнажённая, сидящая на диване | Амедео Модильяни 🇮🇹 | 2010 | Sotheby’s |
| 68.5 | Сидящая женщина | Пабло Пикассо 🇪🇸 | 2016 | Sotheby’s |
| 64.8 | Орел на сосне | Ци Байши 🇨🇳 | 2011 | Другой |
| 64.7 | Супрематическая композиция | Казимир Малевич 🇷🇺 | 2008 | Sotheby’s |
| 64.0 | Мужчины в ее жизни | Энди Уорхол 🇺🇸 | 2010 | Другой |
| 62.3 | 1949-A-No.1 | Клиффорд Стилл 🇺🇸 | 2011 | Sotheby’s |
| 61.8 | Пруд с водяными лилиями | Клод Моне 🇫🇷 | 2008 | Christie’s |
| 61.3 | Чжи Чуань поднимается в гору | Ван Мэн 🇨🇳 | 2011 | Другой |
| 60.4 | Кувшин и фрукты | Поль Сезанн 🇫🇷 | 2019 | Christie’s |
| 58.1 | Номер 19, 1948 | Джексон Поллок 🇺🇸 | 2013 | Christie’s |
| 52.7 | Набросок к портрету папы Иннокентия X | Фрэнсис Бэкон 🇬🇧 | 2007 | Sotheby’s |
| 47.3 | 200 однодолларовых купюр | Энди Уорхол 🇺🇸 | 2009 | Sotheby’s |
| 46.5 | Мужчина с топором | Поль Гоген 🇫🇷 | 2006 | Christie’s |
| 45.2 | Часы с кукушкой и ковер в синих и красных тонах | Анри Матисс 🇫🇷 | 2009 | Christie’s |
| 44.2 | Голова музы | Рафаэль 🇮🇹 | 2009 | Christie’s |
| 34.3 | Венеция: Гранд канал, вид с северо-востока от Палаццо Балби на мост Риальто | Каналетто 🇮🇹 | 2005 | Sotheby’s |
| 25.8 | Прачка | Анри де Тулуз-Лотрек 🇫🇷 | 2005 | Christie’s |
| 25.8 | Дань уважения Матиссу | Марк Ротко 🇺🇸 | 2005 | Christie’s |
| | | | | |
| | | | | *Таблица 2.0* |

<h1><strong><font color="#FF732B">Списки</font></strong></h1>

<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/96321/43.png"></p>

<h3><strong><font color="#809BFF">Задача №1</font></strong></h3>

Сохрани в переменную `chinese_prices` список стоимостей всех картин китайских художников из Таблицы 2.0. Напечатай результат, тип данных, а так же стоимость самой дорогой картины с помощью индекса. Используй f-strings.

In [ ]:
# присвой переменной список стоимостей
chinese_prices = [149.6, 64.8, 61.3]

# напечатай список
print(f'Список стоимостей работ китайских художников: {chinese_prices}')

# тип данных
print(type(chinese_prices))

# самая дорогая картина
print(f'Цена самой дорогой картины китайских художников: {chinese_prices[0]}')

Список стоимостей работ китайских художников: [149.6, 64.8, 61.3]
<class 'list'>
Цена самой дорогой картины китайских художников: 149.6


<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/96337/46.png"></p>

<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/96338/47.png"></p>

<h3><strong><font color="#809BFF">Задача №2</font></strong></h3>

В переменной `american_top_9_prices` сохранен список стоимостей 9 самых дорогих картин американских художников. Используя срезы, напечатайте отдельно 3 самых высоких стоимости , 3 самых низких, медианную стоимость, а также каждую вторую стоимость, начиная с нулевого индекса, во всем наборе данных.

In [ ]:
american_top_9_prices = [117.3, 105.0, 103.4, 95.2, 90.4, 89.9, 87.8, 81.6, 81.3]

# напечатай 3 самых высоких стоимости
print(american_top_9_prices[:3])

# 3 самых низких
print(american_top_9_prices[6:])

# медиану
print(american_top_9_prices[4])

# и каждую вторую стоимость, начиная с нулевого индекса
print(american_top_9_prices[::2])

[117.3, 105.0, 103.4]
[87.8, 81.6, 81.3]
90.4
[117.3, 103.4, 90.4, 87.8, 81.3]


<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/96323/44.png"></p>

<h3><strong><font color="#809BFF">Задача №3</font></strong></h3>

Сохрани в переменную `russian_data` список с именами авторов и стоимостью всех картин российских художников. Напечатай результат, а также длину списка.

In [ ]:
# присвой переменной список
russian_data = ['Казимир Малевич', 64.7]

# напечатай результат
print(russian_data)

# напечатай длину списка
print(len(russian_data))

['Казимир Малевич', 64.7]
2


<h3><strong><font color="#809BFF">Задача №4</font></strong></h3>

В переменной `italian_prices` сохранен список со стоимостью картин всех итальянских художников. Просуммируй все элементы, обратившись к ним через положительный индекс, и напечатай общую стоимость с одним знаком после точки, используя f-strings.

In [ ]:
italian_prices = [478.1, 184.8, 163.0, 69.7, 44.2, 34.3]

# присвой переменной сумму всех элементов
italian_prices_total = italian_prices[0] + italian_prices[1] +italian_prices[2] +italian_prices[3] +italian_prices[4] +italian_prices[5]

print(f'Общая стоимость работ итальянских мастеров: ${italian_prices_total:.1f}')

Общая стоимость работ итальянских мастеров: $974.1


<h1><strong><font color="#FF732B">Цикл for</font></strong></h1>

<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/95324/p.png"></p>

<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/96293/giffycanvas1.gif"></p>

<h3><strong><font color="#809BFF">Задача №5</font></strong></h3>

Напечатай на экране каждую стоимость в списке `italian_prices` с помощью цикла `for`. Придумай для итератора осознанное подходящее название.

In [ ]:
italian_prices = [478.1, 184.8, 163.0, 69.7, 44.2, 34.3]

# задай цикл и напечатай все элементы
for price in  italian_prices:
  print(price)

478.1
184.8
163.0
69.7
44.2
34.3


<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/96317/42.png"></p>

<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/95329/34.png"></p>

<h3><strong><font color="#809BFF">Задача №6</font></strong></h3>

Используя оператор `+=`, присвой с помощью цикла `for` сумму всех значений из списка `french_prices` в переменную `french_prices_total`. 

Помести команду `print(french_prices_total)` сначала в тело цикла, посмотри на результат, а потом закомментируй и вынеси ее из цикла. 

Проанализируй и осознай разницу.

In [ ]:
french_prices = [25.8, 45.2, 46.5, 60.4, 61.8, 88.3, 112.7]
french_prices_total = 0

# задай цикл и просуммируй все элементы
for price in french_prices:
  french_prices_total+=price
print(french_prices_total/len(french_prices))

62.957142857142856


<h1><strong><font color="#FF732B">Изменение значений списка в цикле</font></strong></h1>

<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/96329/45.png"></p>

<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/95869/skip.png"></p>

<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/95342/36.png"></p>

<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/95343/37.png"></p>

<h3><strong><font color="#809BFF">Задача №7</font></strong></h3>

Напиши цикл, который имитирует работу инфляционного калькулятора — то есть пересчитывает прошлую стоимость объектов с учетом инфляции. Цикл должен умножить оригинальную стоимость (все элементы в списке `italian_prices`) на коэффициент `1.1`. В конце отдельно напечатай изменившийся список. Выводите каждый новый результат в формате `Новая скорректированная стоимость: $...`, используя f-strings.

In [ ]:
italian_prices = [450.3, 170.4, 157.2, 61.5, 34.7, 20.0]

# задай цикл и скорректируй стоимость
for price in range(len(italian_prices)):
  italian_prices[price] =  italian_prices[price]*1.1
  print(f'Новая скорректированная стоимость: {italian_prices[price]:.1f}')

# отдельно проверь изменения значений
print(f'\nПроверка изменений: {italian_prices}')

Новая скорректированная стоимость: 495.3
Новая скорректированная стоимость: 187.4
Новая скорректированная стоимость: 172.9
Новая скорректированная стоимость: 67.7
Новая скорректированная стоимость: 38.2
Новая скорректированная стоимость: 22.0

Проверка изменений: [495.33000000000004, 187.44000000000003, 172.92, 67.65, 38.17000000000001, 22.0]


<h3><strong><font color="#809BFF">Задача №8</font></strong></h3>

Напиши цикл, который переводит тип данных в целочисленный для всех значений в списке `italian_prices`, за исключением первого и последних двух. Они должны остаться неизменными. В цикле печатай результат в формате `Целочисленная стоимость: ... млн $`, а в конце проверь изменения.

In [ ]:
italian_prices = [478.1, 184.8, 163.0, 69.7, 44.2, 34.3]

# цикл для изменения типа данных
for price in range(1,len(italian_prices)-2):
  italian_prices[price] = int(italian_prices[price] )
  print(f'Целочисленная стоимость: {italian_prices[price]}')

print(f'\nПроверка изменений: {italian_prices}')

Целочисленная стоимость: 184
Целочисленная стоимость: 163
Целочисленная стоимость: 69

Проверка изменений: [478.1, 184, 163, 69, 44.2, 34.3]


<h1><strong><font color="#FF732B">Сортировка и добавление новых элементов</font></strong></h1>

<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/95366/38.png"></p>

<h3><strong><font color="#809BFF">Задача №9</font></strong></h3>

Используя метод `sort()`, отсортируй все названия картин итальянских художников в алфавитном порядке. Напечатай каждый элемент с помощью цикла.

In [ ]:
italian_titles = [
    'Обнаженная, лежащая на левом боку',
    'Лежащая обнажённая',
    'Обнажённая, сидящая на диване',
    'Спаситель мира',
    'Венеция: Гранд канал, вид с северо-востока от Палаццо Балби на мост Риальто',
    'Голова музы'
    ]

# отсортируй список в алфавитном порядке
italian_titles.sort()

# цикл для печати
for title in italian_titles:
  print(title)

Венеция: Гранд канал, вид с северо-востока от Палаццо Балби на мост Риальто
Голова музы
Лежащая обнажённая
Обнаженная, лежащая на левом боку
Обнажённая, сидящая на диване
Спаситель мира


<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/95368/39.png"></p>

<h3><strong><font color="#809BFF">Задача №10</font></strong></h3>

Используя метод `sort()`, отсортируй все значения в списке `french_prices` по убыванию. Напечатай каждый элемент с помощью цикла.

In [ ]:
french_prices = [46.5, 112.7, 60.4, 25.8, 45.2, 61.8, 88.3]

# отсортируй значения по убыванию
french_prices.sort(reverse=True)

# цикл для печати
for price in french_prices:
  print(price)

112.7
88.3
61.8
60.4
46.5
45.2
25.8


<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/95370/40.png"></p>

<h3><strong><font color="#809BFF">Задача №11</font></strong></h3>

Используя метод `append()`, добавь в список `french_titles` названия пропущенных французских картин. Напечатай список после изменения.

In [ ]:
french_titles = [
    'Прачка',
    'Кувшин и фрукты',
    'Пруд с водяными лилиями',
    'Стога сена'
]

# 3 французские картины из Таблицы 2.0, которых нет в списке:
# Стог сена
# Часы с кукушкой и ковер в синих и красных тонах
# Мужчина с топором


# добавь новые картины
french_titles.append('Стог сена')
french_titles.append('Часы с кукушкой и ковер в синих и красных тонах')
french_titles.append('Мужчина с топором')

# напечатай обновленный список
for title in french_titles:
  print(title)

Прачка
Кувшин и фрукты
Пруд с водяными лилиями
Стога сена
Стог сена
Часы с кукушкой и ковер в синих и красных тонах
Мужчина с топором


<h1><strong><font color="#FF732B">Списки списков (вложенные списки)</font></strong></h1>

<p><img src="https://u.foxford.ngcdn.ru/uploads/inner_file/file/95386/41.png"></p>

<h3><strong><font color="#809BFF">Задача №12</font></strong></h3>

Из вложенного списка `french_dataset` «достань» с помощью двойного индекса следующие значения и напечатай их:

1. Название картины художника, чей рост менее 155 см
2. Стоимость картины Поля Сезанна
3. Имя художника, который написал картину «Стога сена»
4. Стоимость картины «Стог сена»
5. Название картины Анри Матисса

In [ ]:
french_dataset = [
    [112.7, 'Стога сена', 'Клод Моне 🇫🇷'],
    [88.3, 'Стог сена', 'Клод Моне 🇫🇷'],
    [61.8, 'Пруд с водяными лилиями', 'Клод Моне 🇫🇷'],
    [60.4, 'Кувшин и фрукты', 'Поль Сезанн 🇫🇷'],
    [46.5, 'Мужчина с топором', 'Поль Гоген 🇫🇷'],
    [45.2, 'Часы с кукушкой и ковер в синих и красных тонах', 'Анри Матисс 🇫🇷'],
    [25.8, 'Прачка', 'Анри де Тулуз-Лотрек 🇫🇷']
]

# сформируй запрос и напечатай результат
print(french_dataset[6][1])
print(french_dataset[3][0])

Прачка
60.4


<h3><strong><font color="#809BFF">Задача №13</font></strong></h3>

Просуммируй стоимость всех картин в списке списков `spanish_data` с помощью цикла и напечатай результат с одним знаком после точки, используя f-strings.

In [ ]:
spanish_data = [
    [68.5, 'Сидящая женщина', 'Пабло Пикассо 🇪🇸', 'Sotheby\’s'],
    [107.7, 'Обнажённая, зелёные листья и бюст', 'Пабло Пикассо 🇪🇸', 'Christie\’s'],
    [109.7, 'Дора Маар с кошкой', 'Пабло Пикассо 🇪🇸', 'Sotheby\’s'],
    [119.2, 'Молодая девушка с цветочной корзиной', 'Пабло Пикассо 🇪🇸', 'Christie\’s'],
    [194.5, 'Алжирские женщины («Версия O»)', 'Пабло Пикассо 🇪🇸', 'Christie\’s'],
]

# переменная суммы
...

# цикл для суммирования стоимостей
...

print('Общая стоимость картин испанских художников: ${...}')

<h1><strong><font color="#FC4262">Домашняя работа</font></strong></h1>

Вот несколько шагов, чтобы настроить удобную работу над домашними задачами:

1. В сервисе [Google Drive](https://www.google.com/intl/ru/drive/) создай папку `foxford_ds_study` (если еще этого не сделал), в которой ты будешь хранить все учебные материалы с этого курса.
2. Сохрани копию этой тетрадки `Занятие №3.ipynb` *(File → Save a copy in Drive)* в папку `foxford_ds_study`.

Зачем это нужно? 

Файл, в котором работает преподаватель — это «оригинальный» файл и мы не можем работать в нем все вместе, ведь у каждого свой код! Каждому ученику необходимо сделать свою копию файла и выполнять домашку именно в ней. Если хочешь, ты можешь назвать свой файл по имени, например, `Занятие №2 Иван Кнопочкин.ipynb`.

В начале следующего занятия мы обсудим эти задачи вместе.

<h3><strong><font color="#809BFF">Задача №14</font></strong></h3>

Сохрани в переменную `modigliani_prices` список стоимостей всех картин Амедео Модильяни из Таблицы 2.0. Напечатай результат, тип данных, а также стоимость самой дешевой картины с помощью индекса. Используй f-strings.

In [ ]:
# присвой переменной список стоимостей
modigliani_prices = ...

# напечатай список
print('Список стоимостей работ Амедео Модильяни: ...')

# тип данных
...

# самая дешевая картина
...

<div>
  <details>
	  <summary>Подсказка 1/1</summary>
    <p>Для типа данных — используй <code>type()</code>, для самой дешевой картины — срез.</p>
  </details>
</div>

<div>
  <details>
	  <summary>Результат</summary>
    <p>
      Список стоимостей работ Амедео Модильяни: [184.8, 163.0, 69.7]</br>
      class 'list'</br>
      69.7
    </p>
  </details>
</div>

<h3><strong><font color="#809BFF">Задача №15</font></strong></h3>

Используя цикл и функцию `len()`, вычисли среднюю стоимость в списке `french_prices`. Напечатай результат с двумя знаками после точки. Перед выполнением подумай над реализацией в целом, а потом пиши код.

In [ ]:
french_prices = [25.8, 45.2, 46.5, 60.4, 61.8, 88.3, 112.7]

# код решения
...

# напечатай результат
print('Средняя стоимость французских картин: $...')

<div>
  <details>
	  <summary>Подсказка 1/1</summary>
    <p>Среднее значение — это число, равное сумме всех чисел множества, делённое на их количество.</p>
  </details>
</div>

<div>
  <details>
	  <summary>Результат</summary>
    <p>Средняя стоимость французских картин: $62.96</p>
  </details>
</div>

<h3><strong><font color="#809BFF">Задача №16</font></strong></h3>

По неизвестной для рынка искусства причине картины испанских художников подешевели в полтора раза. Напиши цикл, который изменит текущую стоимость работ в переменной `spanish_prices`. Выводи каждый новый результат в формате `Новая сниженная стоимость: $...`, используя f-strings.

In [ ]:
spanish_prices = [68.5, 107.7, 109.7, 119.2, 194.5]

# задай цикл и уцени картины
...

# отдельно проверь изменения значений
print('\nПроверка изменений: ...')

<div>
  <details>
	  <summary>Подсказка 1/3</summary>
    <p>Первоочередная идея — разделить сам итератор на <code>1.5</code> — не пройдет.</p>
  </details>
</div>

<div>
  <details>
	  <summary>Подсказка 2/3</summary>
    <p>Передай диапазон индексов списка <code>italian_prices</code>, используя функции <code>range()</code> и <code>len()</code>.</p>
  </details>
</div>

<div>
  <details>
	  <summary>Подсказка 3/3</summary>
    <p>Меняй каждый элемент списка через индекс (перебором с помощью итератора).</p>
  </details>
</div>

<div>
  <details>
	  <summary>Результат</summary>
    <p>
      Новая сниженная стоимость: $45.7</br>
      Новая сниженная стоимость: $71.8</br>
      Новая сниженная стоимость: $73.1</br>
      Новая сниженная стоимость: $79.5</br>
      Новая сниженная стоимость: $129.7</br>
      </br>
      Проверка изменений: [45.666666666666664, 71.8, 73.13333333333334, 79.46666666666667, 129.66666666666666]
    </p>
  </details>
</div>

<h3><strong><font color="#809BFF">Задача №17</font></strong></h3>

Злоумышленники украли каждую вторую картину, стоимость которых сохранена в списке `french_prices`, и теперь владец коллекции хочет выяснить общую стоимость всех оставшихся картин. 

Напиши 2 цикла — первый должен обесценить каждую вторую картину, а второй — просуммировать стоимость оставшихся в переменную `leftover_paintings_total`.

Напечатай значения списка «после кражи», а также стоимость оставшихся картин в формате: `Стоимость оставшихся картин: $...`.

In [ ]:
french_prices = [25.8, 45.2, 46.5, 60.4, 61.8, 88.3, 112.7]

# код решения
...

<div>
  <details>
	  <summary>Подсказка 1/3</summary>
    <p>Первоочередная идея — обесценить сам итератор — не пройдет.</p>
  </details>
</div>

<div>
  <details>
	  <summary>Подсказка 2/3</summary>
    <p>«Украдена каждая вторая картина» — это значит, что нижняя граница диапазона индексов должна быть <code>1</code> и шаг <code>2</code>. Передай диапазон с помощью функций <code>range()</code> и <code>len()</code>.</p>
  </details>
</div>

<div>
  <details>
	  <summary>Подсказка 3/3</summary>
    <p>Меняй каждый элемент списка через индекс (перебором с помощью итератора).</p>
  </details>
</div>

<div>
  <details>
	  <summary>Результат</summary>
    <p>
      Список стоимостей картин после кражи: [25.8, 0.0, 46.5, 0.0, 61.8, 0.0, 112.7]</br>
      Стоимость оставшихся картин: $246.8
    </p>
  </details>
</div>

<h3><strong><font color="#809BFF">Задача №18</font></strong></h3>

Создай список списков в переменной `chinese_dataset`. В каждом отдельном списке (то есть строке) сохрани следующие значения: `Стоимость картины`, `Название картины`, `Имя китайского художника`. 

Напечатай сначала получившийся результат целиком, а потом закомментируй его и выведи название самой дорогой картины, обратившись к ней с помощью двойного индекса.

In [ ]:
chinese_dataset = [
    ...,
    ...,
    ...
]

# напечатай результат
...

# напечатай название самой дорогой картины
...

<div>
  <details>
	  <summary>Подсказка 1/1</summary>
    <p>В двойном индексе — первый обозначает строку, в которой ищем, а второй — сам элемент.</p>
  </details>
</div>

<div>
  <details>
	  <summary>Результат 1</summary>
    <p>[[149.6, 'Двенадцать ландшафтных сцен', 'Ци Байши 🇨🇳'], [64.8, 'Орел на сосне', 'Ци Байши 🇨🇳'], [61.3, 'Чжи Чуань поднимается в гору', 'Ван Мэн 🇨🇳']]</p>
  </details>
</div>

<div>
  <details>
	  <summary>Результат 2</summary>
    <p>Двенадцать ландшафтных сцен</p>
  </details>
</div>